django/test/utils.py

```diff
@@ -841,6 +845,37 @@ def disable(self):
         setattr(Options, 'default_apps', self.old_apps)
 
 
+class TimeKeeper:
+    def __init__(self):
+        self.records = {}
```

Comment:
```suggestion
        self.records = collections.defaultdict(list)
```

Reply:
This alters the behavour. See https://github.com/django/django/pull/13224#pullrequestreview-462400381 🤔

Reply:
Ah. It is because we are pre-populating `self.records` before the `yield` in the context manager. Changing to `defaultdict` means that the key only gets added after when the context manager is exiting and we do `self.records[name].append(end_time)`.

This means that nested uses of `time_keeper.timed()` - as we do in the database setup -- end up being out of order.